test/e2e/agent_test.go

```diff
@@ -23,47 +23,94 @@ func TestAgent(t *testing.T) {
+			fmt.Println(output)
 			fmt.Println("============ Console output ============")
 			lines := strings.Split(output, "\n")
 			fmt.Println(strings.Join(lines[len(lines)-20:], "\n"))
 			fmt.Println("========================================")
 		})
+
 		It("should have flightctl-agent running", func() {
-			stdout, err := testVM.RunSSH([]string{"sudo", "systemctl", "status", "flightctl-agent"}, nil)
+			stdout, err := harness.VM.RunSSH([]string{"sudo", "systemctl", "status", "flightctl-agent"}, nil)
 			Expect(err).ToNot(HaveOccurred())
 			Expect(stdout.String()).To(ContainSubstring("Active: active (running)"))
 		})
+
+		It("should be reporting tpm info on enrollment request as well as device status", func() {
+			// Get the enrollment Request ID from the console output
+			enrollmentID := harness.GetEnrollmentIDFromConsole()
+			logrus.Infof("Enrollment ID found in VM console output: %s", enrollmentID)
+
+			// Wait for the device to create the enrollment request, and check the TPM details
+			enrollmentRequest := harness.WaitForEnrollmentRequest(enrollmentID)
+			Expect(enrollmentRequest.Spec).ToNot(BeNil())
+			Expect(enrollmentRequest.Spec.DeviceStatus).ToNot(BeNil())
+			Expect(enrollmentRequest.Spec.DeviceStatus.SystemInfo).ToNot(BeNil())
+			Expect(enrollmentRequest.Spec.DeviceStatus.SystemInfo.Measurements).ToNot(BeNil())
+			verifyPCRRegistersNotEmpty(enrollmentRequest.Spec.DeviceStatus.SystemInfo.Measurements)
+
+			// Approve the enrollment and wait for the device details to be populated by the agent
+			harness.ApproveEnrollment(enrollmentID, testutil.TestEnrollmentApproval())
+			logrus.Infof("Waiting for device %s to report status so we can check TPM PCRs again", enrollmentID)
+
+			// wait for the device to pickup enrollment, report measurements on device status
+			Eventually(getDeviceWithStatusSystemInfo, TIMEOUT, POLLING).WithArguments(
+				harness, enrollmentID).ShouldNot(BeNil())
+
+			device := getDeviceWithStatusSystemInfo(harness, enrollmentID)
+
+			// make sure that the PCR registers aren't empty
+			verifyPCRRegistersNotEmpty(device.JSON200.Status.SystemInfo.Measurements)
+
+			// verify that the measurements are the same as the ones we saw in the enrollment request
+			Expect(device.JSON200.Status.SystemInfo.Measurements).To(Equal(enrollmentRequest.Spec.DeviceStatus.SystemInfo.Measurements))
+		})
 	})
 
 })
+
+func verifyPCRRegistersNotEmpty(measurements map[string]string) {
+	Expect(measurements).ToNot(BeNil())
+	for i := 1; i < 10; i++ {
+		pcrReg := fmt.Sprintf("pcr%02d", i)
+		pcr := measurements[pcrReg]
+		Expect(pcr).ToNot(BeNil())
+
+		pcrRemove0x := strings.ReplaceAll(pcr, "0", "")
+		Expect(pcrRemove0x).ToNot(BeEmpty(), "PCR %s is empty or all 0's", pcrReg)
+	}
+}
+
+func getDeviceWithStatusSystemInfo(harness *e2e.Harness, enrollmentID string) *client.ReadDeviceResponse {
+	device, err := harness.Client.ReadDeviceWithResponse(harness.Context, enrollmentID)
```

Comment:
should we log the original error?

Reply:
Hmm, you are right; an error at this point is probably worth logging, at least. At least with the current usage, it should not happen, I believe.

Reply:
Ok, I checked, and I decided to go a little bit harder. The test will fail on errors here. As that should not be happening.